### PR TITLE
Add 'Weather Map' app.

### PR DESCRIPTION
**Pull request by fxb**
### Description

This app displays real-time precipitation radar for a location.

It uses the API provided for free by RainViewer: https://www.rainviewer.com/api.html

Configuration options include:
  * Choose the zoom level, which is expressed in kilometers / miles.
  * Choose from a selection of 8 different color schemes.
  * Show a time overlay in either 12-hour or 24-hour format, or turn it off.
  * Show a coverage overlay using kilometers / miles, or turn it off.
  * Enable or disable the "snow" layer.

##### Original
color_scheme_1

##### Universal Blue (default)
color_scheme_2

##### TITAN
color_scheme_3

##### The Weather Channel (TWC)
color_scheme_4

##### Meteored
color_scheme_5

##### NEXRAD Level III
color_scheme_6

##### Rainbow @ SELEX-IS
color_scheme_7

##### Dark Sky
color_scheme_8

#### Show coverage of the area (width) in kilometers
coverage_km

#### Show coverage of the area (width) in miles
coverage_miles

#### Show snow layer
snow_1

#### 24-hour time format
time_24h

### CODE REVIEW

**Comment by fxb** [comment]
Please do not merge yet, as I want to add truncation of lat / lng for privacy reasons to this app as I saw done here https://github.com/tidbyt/community/pull/181 :)

File: apps/weathermap/weather_map.star [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+            diameter = 3,
+            color = "#f00a",
+        )),
+    ])
+
+def create_image_url(frame, opts):
+    return IMAGE_URL_LAYOUT.format(
+        host = getattr(opts, "host"),
+        path = frame["path"],
+        lat = getattr(opts, "location")["lat"],
```
Comment (betterengineering) [inline apps/weathermap/weather_map.star:225]: Would you mind truncating lat/lng? Mostly, we're concerned about leaking user data to third-party APIs. I think we could get away with truncating to two decimal places?? https://gis.stackexchange.com/questions/8650/measuring-accuracy-of-latitude-and-longitude

**Review by betterengineering — APPROVED** [review]
This is so rad! I'm super excited to put this on my Tidbyt. The one change to truncate user location is all we need.

**Comment by fxb** [comment]
@betterengineering I've added a commit to truncate coordinates. It is adaptive to the selected zoom level, so the more the user zooms out, the more the coordinates get truncated.

**Pull request merged**